File: Leetcode/python/sol3042.py
# leetcode.com/problems/count-prefix-and-suffix-pairs-i/

class Solution:
    def isPrefixAndSuffix(self, str1:str, str2:str) -> bool:
        if len(str1) > len(str2):
            return False
        len_str1 = len(str1)
        return str1 == str2[:len_str1] and str1 == str2[-len_str1:]

    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        rslt = 0
        for i in range(len(words)):
            for j in range(i+1, len(words)):
                if self.isPrefixAndSuffix(words[i], words[j]):
                    rslt+=1
        return rslt

File: Leetcode/cpp/sol3042.cpp

//https://leetcode.com/problems/count-prefix-and-suffix-pairs-i/description/

 class Solution {
public:

    bool isPrefixAndSuffix(const string &str1, const string &str2){
        if (str1.size() > str2.size()){
            return false;
        }
        int size1 = str1.size();
        int size2 = str2.size();
        for(int i = 0; i<size1; i++){
            if (str1[size1-1-i] != str2[size2-1-i] || str1[i] != str2[i]){
                return false;
            }
        }
        return true;
    }

    int countPrefixSuffixPairs(vector<string>& words) {
        int rslt = 0;
        for(int i = 0; i<words.size(); i++){
            for(int j = i + 1; j<words.size(); j++){
                if(isPrefixAndSuffix(words[i], words[j])){
                    rslt++;
                }
            }
        }
        return rslt;
    }
};